bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,556 @@
+            bento_path,
+            updated_deployment_config,
+            s3_bucket_name,
+            region,
+        )
+
+        return ApplyDeploymentResponse(status=Status.OK(), deployment=deployment_pb)
+
+    def describe(self, deployment_pb):
+        try:
+            deployment_spec = deployment_pb.spec
+            ec2_deployment_config = deployment_spec.aws_ec2_operator_config
+            ec2_deployment_config.region = (
+                ec2_deployment_config.region or get_default_aws_region()
+            )
+            if not ec2_deployment_config.region:
+                raise InvalidArgument("AWS region is missing")
+
+            deployment_stack_name = generate_aws_compatible_string(
+                "btml-stack-{namespace}-{name}".format(
+                    namespace=deployment_pb.namespace, name=deployment_pb.name
+                )
+            )
+            try:
+                cf_client = boto3.client("cloudformation", ec2_deployment_config.region)
+                cloudformation_stack_result = cf_client.describe_stacks(
+                    StackName=deployment_stack_name
+                )
+                stack_result = cloudformation_stack_result.get("Stacks")[0]
+
+                if stack_result["StackStatus"] in SUCCESS_CLOUDFORMATION_STACK_STATUS:
+                    if stack_result.get("Outputs"):
+                        outputs = stack_result.get("Outputs")
+                    else:
+                        return DescribeDeploymentResponse(
+                            status=Status.ABORTED('"Outputs" field is not present'),
+                            state=DeploymentState(
+                                state=DeploymentState.ERROR,
+                                error_message='"Outputs" field is not present',
+                            ),
+                        )
+
+                elif stack_result["StackStatus"] in FAILED_CLOUDFORMATION_STACK_STATUS:
+                    state = DeploymentState(state=DeploymentState.FAILED)
+                    return DescribeDeploymentResponse(status=Status.OK(), state=state)
+
+                else:
+                    state = DeploymentState(state=DeploymentState.PENDING)
+                    return DescribeDeploymentResponse(status=Status.OK(), state=state)
+
+            except Exception as error:  # pylint: disable=broad-except
+                state = DeploymentState(
+                    state=DeploymentState.ERROR, error_message=str(error)
+                )
+                return DescribeDeploymentResponse(
+                    status=Status.INTERNAL(str(error)), state=state
+                )
+
+            outputs = {o["OutputKey"]: o["OutputValue"] for o in outputs}
+            info_json = {}
+
+            if "S3Bucket" in outputs:
+                info_json["s3_bucket"] = outputs["S3Bucket"]
+
+            state = DeploymentState(
+                state=DeploymentState.RUNNING, info_json=json.dumps(info_json)
+            )
```

Comment:
I think we should have the state to `RUNNING` only when the EC2 instances are running, instead of the stack status. That way, users have a clear indicator of the deployment running or not and at the same time, they can take advantage the `wait` parameter for deploy/update deployments

Reply:
What if there are multiple insatnces spawned by scaling group,and they are different states?

Reply:
That's a good question.
I think `RUNNING` for deployment means the deployment is available to take and process requests. 
If one instance is running among the group, we should mark the deployment as running.

Users can find individual instance's status from the `info_json` where we keep all of the describe_instances' result.

Reply:
fixed